igor-web/src/main/java/com/netflix/spinnaker/igor/gcb/GoogleCloudBuildCache.java

```diff
@@ -62,12 +65,27 @@ private void internalUpdateBuild(String buildId, String status, String build) {
     redisClientDelegate.withCommandsClient(c -> {
       String oldStatus = c.hget(key, "status");
       if (allowUpdate(oldStatus, status)) {
+        int ttlSeconds = getTtlSeconds(status);
         c.hset(key, "status", status);
         c.hset(key, "build", build);
+        c.expire(key, ttlSeconds);
       }
     });
   }
 
+  private int getTtlSeconds(String statusString) {
+    try {
+      GoogleCloudBuildStatus status = GoogleCloudBuildStatus.valueOf(statusString);
+      if (status.isComplete()) {
+        return completedTtlSeconds;
+      } else {
+        return inProgressTtlSeconds;
+      }
+    } catch (IllegalArgumentException e) {
+      return inProgressTtlSeconds;
```

Comment:
Not a big fan of swallowing exceptions. Is this something that we can log?

Reply:
We're swallowing a very targeted exception, where the status on the received GCB build isn't one we know about---as if GCB adds a new status, I'd rather the failure mode be less efficient caching than actual user-facing breakage.

It does seem reasonable to log this since if it happens we'd want to update igor to know of the new status, so I'll add that in.